dbt/adapters/bigquery/column.py

```diff
@@ -252,8 +268,23 @@ def _format_nested_data_type(unformatted_nested_data_type: Union[str, Dict[str,
     if isinstance(unformatted_nested_data_type, str):
         return unformatted_nested_data_type
     else:
+        parent_data_type = unformatted_nested_data_type.pop(_PARENT_DATA_TYPE_KEY, None)
```

Comment:
I think this could be clearer and more self-documented by unpacking and defaulting to an empty string (avoiding the double check on line 285 and allowing for the unpacking). For example:
```python
parent_data_type, *parent_constraints = unformatted_nested_data_type.pop(_PARENT_DATA_TYPE_KEY, "").split()

formatted_nested_types = [
    f"{column_name} {_format_nested_data_type(column_type)}"
    for column_name, column_type in unformatted_nested_data_type.items()
]

formatted_nested_type = f"""struct<{", ".join(formatted_nested_types)}>"""

if parent_data_type.lower() == "array":
    formatted_nested_type = f"""array<{formatted_nested_type}>"""

if parent_constraints:
    formatted_nested_type = f"""{formatted_nested_type} {" ".join(parent_constraints)}"""
```

Reply:
love this suggestion, absolutely agree. thank you!

Reply:
for posterity: it ended up needing a [little](https://github.com/dbt-labs/dbt-bigquery/pull/806/files#diff-c9d0e58b6736ccad3debb75fdfd3973a75eaa9114c2afbc65cd74d7b048de575R278) [more](https://github.com/dbt-labs/dbt-bigquery/pull/806/files#diff-c9d0e58b6736ccad3debb75fdfd3973a75eaa9114c2afbc65cd74d7b048de575R291) tweaking but overall much nicer to look at!